### PR TITLE
[UXIT-1537] Event Page · Add Speakers section

### PR DESCRIPTION
**Pull request by mirhamasala**
## 📝 Description

This PR introduces the "Speakers" feature for events, allowing event organizers to display speaker information on event pages. The most important changes include adding the speakers schema, updating the event front matter schema, and modifying the event page to display speaker information.

- **Type:** New feature

## 🛠️ Key Changes

- **Extend Event with speakers:** Updated the event data model to include a list of speakers.
- **Fix type:** Corrected type definitions to ensure type safety when working with the speaker data.
- **Show speakers:** Implemented UI changes to display the speakers section on the event page.
- **Add speakers to FIL Bangkok 2024 event:** Added actual speaker data for the FIL Bangkok 2024 event.
- **Fix layout:** Adjusted the layout to accommodate the speakers section and improve overall design.

Schema and Configuration Updates:

* [`src/app/_schemas/event/SpeakerSchema.ts`](diffhunk://#diff-d8fb6671be7f329e4e8760f525ce13ea10d0963298376fdcdf3b1ce950cca7afR1-R15): Added `SpeakersSchema` to define the structure of speaker information, including fields for name, title, company, LinkedIn URL, and image.
* [`src/app/_schemas/event/FrontMatterSchema.ts`](diffhunk://#diff-e6a8563849805d6c70c46cb94c10d42295786e38a4c2f7dff9bdd78ddc9f8bb4R36): Extended the front matter schema to include an optional `speakers` field.
* [`public/admin/config.yml`](diffhunk://#diff-c4b7720f4813ef734de91022075db713c78b4d0566dd7a95f868d75a65935486R726-R754): Added a "Speakers" section to the admin configuration to allow input of speaker information.

Component and Page Updates:

* [`src/app/_components/KeyMemberCard.tsx`](diffhunk://#diff-03f882b50e160e326a19d1fe55f5f4acffb62fab3d96142e22b08c95e678910bR1-R38): Modified the `KeyMemberCard` component to handle both `MemberData` and `Speaker` types, and updated the image handling logic.
* `src/app/events/[slug]/page.tsx`: Updated the event page to display a section for speakers, using the `KeyMemberCard` component within a `CardGrid`.

Data Handling:

* [`src/app/_utils/convertMarkdownToEventData.ts`](diffhunk://#diff-29518a0532b55abfce2ff4c23c0c25beacc1380c626b1d3ab532ed14ffd40d0aR25): Updated the conversion function to include the `speakers` field from markdown data.

## 📌 To-Do Before Merging

- [x] Verify speaker data is correctly displayed.
- [x] Ensure layout changes work across different screen sizes (responsive).

## 🧪 How to Test

- **Setup:** Ensure the latest event data and speakers are available.
- **Steps to Test:**
  1. Navigate to the FIL Bangkok 2024 event page.
  2. Verify that the list of speakers is correctly displayed with their names, titles, and company information.
  3. Test the layout on both desktop and mobile views to ensure responsiveness.
  4. Verify that speaker images are correctly loaded and displayed with the right aspect ratio.
  
- **Expected Results:** 
  - The speakers should appear correctly in the designated section of the event page.
  - The layout should be responsive and look good across different screen sizes.
  - No layout or rendering issues should be present.

## 📸 Screenshots
CleanShot 2024-09-27 at 10 58 22@2x

### CODE REVIEW

File: public/admin/config.yml [hunk 1]
```diff
@@ -723,6 +723,35 @@ collections:
             required: false
             hint: Paste the embed URL for your Luma event here to display the events directly on the page. Include relevant tags when appropriate, e.g., https://lu.ma/embed/calendar/cal-nlDvL4B7Ko1swF0/events?lt=light&tag=FIL%20Bangkok%202024.
       - *image_config
+      - name: "speakers"
+        label: "Speakers"
+        widget: "list"
+        required: false
+        fields:
+          - name: "name"
+            label: "Name"
```
Comment (barbaraperic) [inline public/admin/config.yml:732]: Maybe Full Name?

```suggestion
            label: "Full Name"
```

File: public/admin/config.yml [hunk 2]
```diff
@@ -723,6 +723,35 @@ collections:
...
+            widget: "string"
+          - name: "title"
+            label: "Title"
```
Comment (barbaraperic) [inline public/admin/config.yml:735]:
```suggestion
            label: "Job Title"
```

File: src/app/_components/KeyMemberCard.tsx [hunk 3]
```diff
@@ -1,34 +1,41 @@
...
-      <StaticImage
-        data={image}
+      <Image
+        src={image.src}
         alt={`Photo of ${name}`}
         sizes="150px"
         className="aspect-[3/4] w-32 rounded object-cover"
+        {...(isStaticImage
+          ? { placeholder: 'blur' }
+          : { width: 150, height: 200 })}
```
Comment (barbaraperic) [inline src/app/_components/KeyMemberCard.tsx:27]: Curious about this one? Do we needed if images are local assets?

File: src/app/_components/KeyMemberCard.tsx [hunk 4]
```diff
@@ -1,34 +1,41 @@
...
 
       <div className="m-3 grow">
         <Heading tag="h3" variant="lg">
           {name}
         </Heading>
 
-        <p className="mt-1 text-brand-300">{title}</p>
+        <p className="mt-1 text-brand-300">
+          {title}
+          {'company' in props && props.company && `, ${props.company}`}
```
Comment (barbaraperic) [inline src/app/_components/KeyMemberCard.tsx:37]: I might be missing something - why the check? isn't company required?

File: src/app/_components/KeyMemberCard.tsx (hunk 3 above)
Comment (mirhamasala) [reply src/app/_components/KeyMemberCard.tsx:27]: For speakers, they are being imported dynamically via the CMS.

File: src/app/_components/KeyMemberCard.tsx (hunk 4 above)
Comment (mirhamasala) [reply src/app/_components/KeyMemberCard.tsx:37]: For `Speakers` it is, but for for `MemberData` (used for Filecoin Foundation Board members, for example, on the company page, it isn't.

**Pull request merged**